From cade63fba570ffb7381f5e24350fddd41c64c0f4 Mon Sep 17 00:00:00 2001
From: Ben Beasley <code@musicinmybrain.net>
Date: Tue, 15 Oct 2024 15:41:10 -0400
Subject: [PATCH] Allow Starlette 0.40.x (#5000)

This release of Starlette contains a fix for a security bug:

- GHSA-f96h-pmfr-66vw:
  https://github.com/encode/starlette/security/advisories/GHSA-f96h-pmfr-66vw
- CVE-2024-47874: https://nvd.nist.gov/vuln/detail/CVE-2024-47874
---
 platformio/dependencies.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platformio/dependencies.py b/platformio/dependencies.py
index 0a81f868cf..e358453383 100644
--- a/platformio/dependencies.py
+++ b/platformio/dependencies.py
@@ -43,7 +43,7 @@ def get_pip_dependencies():
     home = [
         # PIO Home requirements
         "ajsonrpc == 1.2.*",
-        "starlette >=0.19, <0.40",
+        "starlette >=0.19, <0.41",
         'uvicorn == 0.16.0; python_version < "3.7"',
         'uvicorn >=0.16, <0.31; python_version >= "3.7"',
         "wsproto == 1.*",
